### PR TITLE
Add default sort field to o2m interface

### LINKED ISSUE #18076: Allow configuring non-manual sort order for the o2m style interfaces
### Discussed in https://github.com/directus/directus/discussions/12771

<div type='discussions-op-text'>

<sup>Originally posted by **paulboudewijn** November 19, 2021</sup>
When no sort field is configured for a relation, the PK is used as the default sort field instead of using the field(s) used in the display template. This results in a seemingly random list of items in the m2m interface.

I have a PR ready.

### To Reproduce

1. Create a 'book' collection with the 'title' and 'subtitle' fields
2. Create an 'author' collection
3. Create a m2m relation between 'author' and 'book', without sortfield and interface displaytemplate
4. Note that the m2m interface list 'books' on the 'author' collection is sorted by the PK field
5. Add displaytemplate `{{book_id.title}} - {{book_id.subtitle}}` to the interface
6. Note that the m2m interface list is still sorted in the PK field, although title and subtitle alphabetically could suggest otherwise
7. Add a sort field  `sort` to the relation
8. Note that the interface list is sorted on this column (as expected)</div>

### PR DESCRIPTION
**Pull request by alexvdvalk**
## Scope

What's changed:

- Add sort and sort direction fields to the o2m interface config. 
- When the interface loads, it will apply this sort to the related items, as long as the relation does not have a manual sort field
- If there is a sort field specified, this default sort is ignored.

## Potential Risks / Drawbacks

- if a manual sort field is setup, the new fields are ignored and this may be a bit confusing

## Review Notes / Questions

- Is there a better naming convention here? A o2m can have a 'sort' field, but this pr refers to sorting the items by another field. ie. alphabetically.
- Directus sorting factors in Upper and lower case, so user may get unexpected results when applying a default sort, but this is consistent with sorting inside the table layout.

---

Discussion: https://github.com/directus/directus/discussions/22799

### CODE REVIEW

**Comment by alexvdvalk** [comment]
Demo:

https://github.com/user-attachments/assets/4e9193b9-3d9b-4a03-8669-0dc6b68e9b1f

File: app/src/interfaces/list-o2m/list-o2m.vue [hunk 1]
```diff
@@ -39,6 +39,8 @@ const props = withDefaults(
 		enableSearchFilter?: boolean;
 		enableLink?: boolean;
 		limit?: number;
+		sort?: string;
+		sort_direction: '+' | '-' | undefined;
```
Comment (hanneskuettner) [inline app/src/interfaces/list-o2m/list-o2m.vue:43]: Let's make this camelCase, just like all other props/options

File: app/src/interfaces/list-o2m/list-o2m.vue [hunk 2]
```diff
@@ -111,6 +113,11 @@ const query = computed<RelationQueryMultiple>(() => {
 		return q;
 	}
 
+	if (props.sort && !relationInfo.value?.sortField) {
+		const direction = props.sort_direction ? props.sort_direction : '';
```
Comment (hanneskuettner) [inline app/src/interfaces/list-o2m/list-o2m.vue:117]:
```suggestion
		const direction = props.sort_direction ?? '';
```

**Comment by hanneskuettner** [comment]
The existing `sort` `ref` should probably be renamed to `manualSort` in order to avoid confusion with the now introduced `sort` prop (eslint is also complaining)

File: app/src/interfaces/list-o2m/list-o2m.vue [hunk 3]
```diff
@@ -111,6 +113,11 @@ const query = computed<RelationQueryMultiple>(() => {
...
+		q.sort = [direction + props.sort];
```
Comment (hanneskuettner) [inline app/src/interfaces/list-o2m/list-o2m.vue:118]: Let's align that with the manual sort string creation
```suggestion
		q.sort = [`${props.sort_direction ?? ''}${props.sort}`];
```

**Comment by alexvdvalk** [comment]
I'm going to extens this to support the m2m interface too.

**Comment by hanneskuettner** [comment]
After having tried to implement a solution for M2M interfaces with @alexvdvalk we realized that there apparently is no system interface that allows the selection of a single nested field in a collection (and that `system-field` does not support the interaction). This is enough reason for me to scrap the M2M support for now, but get the O2M one merged now, and revisit the extension to M2M interfaces at a later point in time.

**Review by hanneskuettner — APPROVED** [review]
LGTM 🎸 Thank you @alexvdvalk 

Let's revisit the M2M interface at a later point in time

**Pull request merged**